org.scala-ide.sdt.core.tests/src/org/scalaide/core/compiler/NamePrinterTest.scala

```diff
@@ -387,6 +387,121 @@ class NamePrinterTest {
+         |}""",
+      "MethodcallInNestedStructure.method")
+  }
+
+  @Test
+  def testWithNamedParamCtorCallInMethod() {
+    testWith(
+      """|package test.named.param.ctor.call
+         |class NamedParamCtorCallInMethod(arg: String) {
+         |  def method {
+         |    new NamedParamCtorCallInMethod(arg/**/ = "aaahrg")
+         |  }
+         |}""",
+      "test.named.param.ctor.call.NamedParamCtorCallInMethod(arg: String)")
+  }
+
+  @Test
+  def testWithSimpleAnnotation() {
+    testWith(
+      """|import scala.annotation._
+         |class SimpleAnnotation extends StaticAnnotation
+         |@SimpleAnnotation/**/
+         |class WithSimpleAnnotation
+      """,
+      "SimpleAnnotation")
+  }
+
+  @Test
+  def testWithAnnotationOnDefOnParam() {
+    testWith(
+      """|import scala.annotation._
+         |class AnnotationOnDefOnParam(val name: String) extends StaticAnnotation
+         |object AnnotationOnDefOnParam {
+         |  @AnnotationOnDefOnParam(name/**/ = "Hallo")
+         |  def test = 32
+         |}""",
+      "AnnotationOnDefOnParam(name: String)")
+  }
+
+  @Test
+  def testWithAnnotationOnDef() {
+    testWith(
+      """|import scala.annotation._
+         |class AnnotationOnDef(val name: String) extends StaticAnnotation
+         |object AnnotationOnDef {
+         |  @AnnotationOnDef/**/(name = "Hallo")
+         |  def test = 32
+         |}""",
+      "AnnotationOnDef")
+  }
+
+  @Test
+  def testWithJavaxGeneratedAnnotationOnClass() {
+    testWith(
+      """|import javax.annotation._
+         |@Generated/**/(Array("today"))
+         |class WithJavaxGeneratedAnnotationOnClass""",
+      "javax.annotation.Generated")
+  }
+
+  @Test
+  def testWithJavaxGeneratedAnnotationOnMethod() {
+    testWith(
+      """|import javax.annotation._
+         |class WithJavaxGeneratedAnnotationOnMethod {
+         |  @Generated/**/(Array("yesterday"))
+         |  def method = Unit
+         |}""",
+      "javax.annotation.Generated")
+  }
+
+  @Test
+  def testWithJavaxResourceAnnotationOnMethod() {
+    testWith(
+      """|import javax.annotation._
+         |class WithJavaxResourceAnnotationOnResource {
+         |  @Resource/**/(name = "test", lookup = "anywhere")
+         |  def method = Unit
+         |}""",
+      "javax.annotation.Resource")
+  }
+
+  @Test
+  def testWithAxillaryCtor() {
+    testWith(
+      """|class WithAxillaryCtor(str: String) {
+         |  def this(i: Int) = this(i.toString)
+         |  def foo() { new WithAxillaryCtor(/**/33) }
+         |}""",
+      "WithAxillaryCtor(i: Int)")
+  }
+
+  @Test
+  def testWithMethodCallOnGeneric() {
+    testWith(
+      """|class WithMethodCallOnGeneric[A](a: A) {
+         |  def method[B, C](b: B, c: C) = b.hashCode + c.hashCode
+         |  def foo = {
+         |    val tst = new WithMethodCallOnGeneric("string")
+         |    tst.method(/**/1L, 0.0)
+         |  }
+         |}""",
+      "WithMethodCallOnGeneric.method[B, C](b: B, c: C)")
```

Comment:
Maybe because the tree is different from the case of simple method application. A generic call is

```
Apply(
  TypeApply(Select(..., "method"), List(B, C)))
  List(b, c))
```